From rahul.ruikar@gmail.com  Thu Oct  7 10:51:06 2010
From: Rahul Ruikar <rahul.ruikar@gmail.com>
To: Greg Kroah-Hartman <gregkh@suse.de>, Tejun Heo <tj@kernel.org>,
	asanchez@neurowork.net, rjw@sisk.pl, csanchez@neurowork.net
Cc: linux-usb@vger.kernel.org, linux-kernel@vger.kernel.org,
	Rahul Ruikar <rahul.ruikar@gmail.com>
Subject: usb: core: endpoint: Fix error path
Date: Thu,  7 Oct 2010 09:31:12 +0530
Message-Id: <1286424072-2819-1-git-send-email-rahul.ruikar@gmail.com>

In function usb_create_ep_devs()
call put_device() when device_register() fails.

Signed-off-by: Rahul Ruikar <rahul.ruikar@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/endpoint.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/usb/core/endpoint.c
+++ b/drivers/usb/core/endpoint.c
@@ -202,7 +202,7 @@ int usb_create_ep_devs(struct device *pa
 	return retval;
 
 error_register:
-	kfree(ep_dev);
+	put_device(&ep_dev->dev);
 exit:
 	return retval;
 }
